src/System.Private.CoreLib/shared/System/Runtime/Intrinsics/Arm/AdvSimd.PlatformNotSupported.cs

```diff
@@ -75,6 +75,210 @@ public new abstract class Arm64 : ArmBase.Arm64
             ///   A64: RBIT Vd.16B, Vn.16B
             /// </summary>
             public static Vector128<sbyte> ReverseElementBits(Vector128<sbyte> value) { throw new PlatformNotSupportedException(); }
+
+            /// <summary>
+            /// uint8x8_t vzip2_u8(uint8x8_t a, uint8x8_t b)
+            ///   A64: ZIP2 Vd.8B, Vn.8B, Vm.8B
+            /// </summary>
+            public static Vector64<byte> ZipHigh(Vector64<byte> left, Vector64<byte> right) { throw new PlatformNotSupportedException(); }
```

Comment:
It looks like ARM32 has `VZIP`, how does it differ from `ZIP1`/`ZIP2` exposed on ARM64?

Reading, it looks like `ZIP1` gets the lower interleaved elements and `ZIP2` gets the upper interleaved elements.

It looks like `VZIP` does the same, but in a single instruction and reuses the two input registers, is that correct?

Reply:
Yeah that's correct, on `VZIP` both registers are destructive, so to get the same behaviour one of the arguments has to to be duplicated before the call to preserve the original value.

I wasn't quite sure how to handle this so in the design proposal I had separated them out and had them treated them as we did in the C compiler as different intrinsics.